python-checks/src/main/java/org/sonar/python/checks/hotspots/SQLQueriesCheck.java

```diff
@@ -103,36 +93,55 @@ private boolean isSQLQueryFromDjangoDBConnection(String functionName) {
     return isUsingDjangoDBConnection && functionName.equals("execute");
   }
 
-  @Override
-  public void visitNode(SubscriptionContext context) {
-    CallExpression callExpressionTree = (CallExpression) context.syntaxNode();
-    if(callExpressionTree.callee().is(Tree.Kind.QUALIFIED_EXPR)) {
-      String functionName = ((QualifiedExpression) callExpressionTree.callee()).name().name();
-      if ((isSQLQueryFromDjangoModel(functionName) || isSQLQueryFromDjangoDBConnection(functionName)) && !isException(callExpressionTree, functionName)) {
-        context.addIssue(callExpressionTree, MESSAGE);
+  private void checkCallExpression(SubscriptionContext context) {
+    CallExpression callExpression = (CallExpression) context.syntaxNode();
+    List<Tree> secondaryLocations = new ArrayList<>();
+
+    Symbol symbol = callExpression.calleeSymbol();
+    if (symbol != null && "django.db.models.expressions.RawSQL".equals(symbol.fullyQualifiedName())
+      && isSensitive(callExpression, secondaryLocations)) {
+
+      addIssue(context, callExpression, secondaryLocations);
+      return;
+    }
+
+    if (callExpression.callee().is(Tree.Kind.QUALIFIED_EXPR)) {
+      String functionName = ((QualifiedExpression) callExpression.callee()).name().name();
+      if ((isSQLQueryFromDjangoModel(functionName) || isSQLQueryFromDjangoDBConnection(functionName))
+        && !isException(callExpression, functionName) && isSensitive(callExpression, secondaryLocations)) {
+
+        addIssue(context, callExpression, secondaryLocations);
       }
     }
-    super.visitNode(context);
+  }
+
+  private static void addIssue(SubscriptionContext context, CallExpression callExpression, List<Tree> secondaryLocations) {
+    PreciseIssue preciseIssue = context.addIssue(callExpression, MESSAGE);
+    secondaryLocations.forEach(secondary -> preciseIssue.secondary(secondary, null));
   }
 
   private static boolean isException(CallExpression callExpression, String functionName) {
     List<Argument> argListNode = callExpression.arguments();
     if (extraContainsFormattedSqlQueries(argListNode, functionName)) {
       return false;
     }
+    return argListNode.isEmpty();
+  }
+
+  private static boolean isSensitive(CallExpression callExpression, List<Tree> secondaryLocations) {
```

Comment:
I believe that this method has a strange signature:
* It both returns something and has side effects on its parameters
* If I understand correctly, `secondaryLocations` can have at most 1 element, so having a list is misleading and makes this method hard to understand

I would change that to: 
`private static Optional<Tree> sensitiveArgumentValue(CallExpression callExpression) {`